library/Ivoz/Provider/Domain/Model/SurvivalDevice/SurvivalDevice.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Ivoz\Provider\Domain\Model\SurvivalDevice;
+
+use Assert\Assertion;
+
+/**
+ * SurvivalDevice
+ */
+class SurvivalDevice extends SurvivalDeviceAbstract implements SurvivalDeviceInterface
+{
+    use SurvivalDeviceTrait;
+
+    /**
+     * @codeCoverageIgnore
+     * @return array<string, mixed>
+     */
+    public function getChangeSet(): array
+    {
+        return parent::getChangeSet();
+    }
+
+    /**
+     * Get id
+     * @codeCoverageIgnore
+     */
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    protected function sanitizeValues(): void
+    {
+        $this->sanitizeProxyLogic();
+    }
+
+    public function setUdpPort(int $udpPort): static
+    {
+        Assertion::between($udpPort, 1, 65535, "Invalid udp port number: $udpPort. Must be between 1 and 65535");
+        return parent::setUdpPort($this->sanitizePort($udpPort));
+    }
+
+    public function setTcpPort(int $tcpPort): static
+    {
+        Assertion::between($tcpPort, 1, 65535, "Invalid tcp port number: $tcpPort. Must be between 1 and 65535");
+        return parent::setTcpPort($this->sanitizePort($tcpPort));
+    }
+
+    public function setTlsPort(int $tlsPort): static
+    {
+        Assertion::between($tlsPort, 1, 65535, "Invalid tls port number: $tlsPort. Must be between 1 and 65535");
+        return parent::setTlsPort($this->sanitizePort($tlsPort));
+    }
+
+    public function setWssPort(int $wssPort): static
+    {
+        Assertion::between($wssPort, 1, 65535, "Invalid wss port number: $wssPort. Must be between 1 and 65535");
+        return parent::setWssPort($this->sanitizePort($wssPort));
+    }
+
+    public function setProxy(string $proxy): static
+    {
+        $proxy = trim($proxy);
+        if (empty($proxy)) {
+            throw new \DomainException('Proxy cannot be empty');
+        }
+
+        return parent::setProxy($proxy);
+    }
+
+    protected function sanitizeProxyLogic(): void
+    {
+        $outboundProxy = $this->getOutboundProxy();
+
+        if ($outboundProxy !== null) {
+            $outboundProxy = trim($outboundProxy);
+
+            if (empty($outboundProxy)) {
+                throw new \DomainException('Outbound Proxy cannot be empty.');
+            }
+
+            if ($this->getProxy() && strpos($this->getProxy(), ':') !== false) {
+                throw new \DomainException('Proxy must not include a port when Outbound Proxy is set.', 70003);
+            }
+        } else {
+            $this->setOutboundProxy(null);
+        }
+    }
```

Comment:
```suggestion
    protected function sanitizeProxyLogic(): void
    {
        $outboundProxy = $this->getOutboundProxy();

        if (is_null($outboundProxy)) {
            return;
        }
        
        $emptyOutboundProxy = empty(
            trim($outboundProxy)
        );

        if ($emptyOutboundProxy) {
            throw new \DomainException('Outbound Proxy cannot be empty.');
        }

        if ($this->getProxy() && strpos($this->getProxy(), ':') !== false) {
            throw new \DomainException('Proxy must not include a port when Outbound Proxy is set.', 70003);
        }
    }
```

- Choose guards (condition -> return) over if/else blocks
- _$this->setOutboundProxy(null);_ only gets called if OutboundProxy is already null.